internal/client/adapter/client_adapter.go

```diff
@@ -556,6 +758,138 @@ func (ca *ClientAdapter[H, Hasher, N, E, Header]) SetBlockTree(blocktree *blockt
 	panic("unimplemented")
 }
 
+// StorageProvider impl
```

Comment:
why does the `ClientAdapter` need to implement `StorageProvider`?  The intention was to implement it for the `Client`, and use methods from `StorageProvider` to implement these `StorageState` methods.